uaclient/entitlements/esm.py

```diff
@@ -22,15 +22,30 @@ class ESMAppsEntitlement(ESMBaseEntitlement):
 
     @property
     def repo_pin_priority(self) -> "Optional[str]":
-        """Only Xenial esm-apps should perform repo pinning"""
-        if "xenial" == util.get_platform_info()["series"]:
-            return "never"
-        return None  # No pinning on >= trusty
+        """All LTS with the exception of Trusty should pin esm-apps."""
+        series = util.get_platform_info()["series"]
+        if series == "trusty":
+            return None
+        config_allow_beta = util.is_config_value_true(
+            config=self.cfg.cfg, path_to_value="features.allow_beta"
+        )
+        if config_allow_beta or self.is_beta is False:
+            if util.is_lts(series):
+                return "never"
+        return None
 
     @property
     def disable_apt_auth_only(self) -> bool:
         """Only xenial esm-apps should remove apt auth files upon disable"""
```

Comment:
I think this comment needs updating to something like
```suggestion
        """All LTS with the exception of Trusty should remove apt auth files upon disable"""
```